[libjpeg-turbo] find_package fails when built statically

****

**Is your feature request related to a problem? Please describe.**

My team likes to statically link everything on Windows, so we build our vcpkg dependencies using the following custom triplet file (saved as _path/to/my/triplets/x64-windows.cmake_):

```cmake
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE static)
set(VCPKG_LIBRARY_LINKAGE static)
```

Then, we consume libjpeg-turbo in our CMake project as follows:

```cmake
find_package(libjpeg-turbo REQUIRED)

add_executable(tjexample tjexample.c)

target_link_libraries(tjexample PRIVATE libjpeg-turbo::turbojpeg-static)
```

However, the `find_package` call fails with the following error:

```
CMake Error at D:/Source/vcpkg/installed/x64-windows/share/libjpeg-turbo/libjpeg-turboTargets.cmake:80 (message):
  The imported target "libjpeg-turbo::turbojpeg-static" references the file

     "D:/Source/vcpkg/installed/x64-windows/debug/lib/turbojpeg-static.lib"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "D:/Source/vcpkg/installed/x64-windows/share/libjpeg-turbo/libjpeg-turboTargets.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  D:/Source/vcpkg/installed/x64-windows/share/libjpeg-turbo/libjpeg-turboConfig.cmake:27 (include)
  D:/Source/vcpkg/scripts/buildsystems/vcpkg.cmake:800 (_find_package)
  CMakeLists.txt:5 (find_package)
```

libjpeg-turbo does indeed install a file named _turbojpeg-static.lib_, yet for some reason the portfile is removing the _-static_ suffix after installation. As a result, there is a mismatch between the files on disk and the generated CMake targets.

https://github.com/microsoft/vcpkg/blob/5962cf5a3bfc2124e8de991a4e45a8edcb3b4c69/ports/libjpeg-turbo/portfile.cmake#L58-L70

**Proposed solution**

I propose removing the lines mentioned above. However, I don't fully understand why those lines were added in the first place.

Adding a few people who might be able to shed some light: @drdanz, @dg0yt 

**Describe alternatives you've considered**

It is possible to work around this issue by manually passing the appropriate values to `target_include_directories` and `target_link_libraries`.

--- a/ports/libjpeg-turbo/portfile.cmake
+++ b/ports/libjpeg-turbo/portfile.cmake
@@ -54,6 +54,10 @@ vcpkg_cmake_configure(
 )
 
 vcpkg_cmake_install()
+vcpkg_copy_pdbs()
+
+vcpkg_fixup_pkgconfig()
+vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/libjpeg-turbo)
 
 # Rename libraries for static builds
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
@@ -64,13 +68,23 @@ if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/lib/jpeg-static.lib")
         file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/jpeg-static.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/jpeg.lib")
         file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/turbojpeg-static.lib" "${CURRENT_PACKAGES_DIR}/debug/lib/turbojpeg.lib")
+    endif()
     
-    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
+    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
+    
+    if (EXISTS "${CURRENT_PACKAGES_DIR}/share/${PORT}/libjpeg-turboTargets-debug.cmake")
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/libjpeg-turboTargets-debug.cmake"
+            "jpeg-static${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}" "jpeg${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}")
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/libjpeg-turboTargets-debug.cmake"
+            "turbojpeg-static${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}" "turbojpeg${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}")
+    endif()
+    if (EXISTS "${CURRENT_PACKAGES_DIR}/share/${PORT}/libjpeg-turboTargets-release.cmake")
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/libjpeg-turboTargets-release.cmake"
+            "jpeg-static${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}" "jpeg${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}")
+        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/share/${PORT}/libjpeg-turboTargets-release.cmake"
+            "turbojpeg-static${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}" "turbojpeg${VCPKG_TARGET_STATIC_LIBRARY_SUFFIX}")
     endif()
 endif()
-
-vcpkg_fixup_pkgconfig()
-vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/libjpeg-turbo)
 
 file(REMOVE_RECURSE
      "${CURRENT_PACKAGES_DIR}/debug/share"
@@ -81,5 +95,3 @@ file(COPY "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/vcpkg-cmake-wrapper.cmake" DESTINATION "${CURRENT_PACKAGES_DIR}/share/jpeg")
 
 file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
-
-vcpkg_copy_pdbs()

--- a/ports/libjpeg-turbo/vcpkg.json
+++ b/ports/libjpeg-turbo/vcpkg.json
@@ -1,8 +1,10 @@
 {
   "name": "libjpeg-turbo",
   "version": "2.1.3",
+  "port-version": 1,
   "description": "libjpeg-turbo is a JPEG image codec that uses SIMD instructions (MMX, SSE2, NEON, AltiVec) to accelerate baseline JPEG compression and decompression on x86, x86-64, ARM, and PowerPC systems.",
   "homepage": "https://github.com/libjpeg-turbo/libjpeg-turbo",
+  "license": "BSD-3-Clause",
   "dependencies": [
     {
       "name": "vcpkg-cmake",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3630,7 +3630,7 @@
     },
     "libjpeg-turbo": {
       "baseline": "2.1.3",
-      "port-version": 0
+      "port-version": 1
     },
     "libjuice": {
       "baseline": "0.9.6",

--- a/versions/l-/libjpeg-turbo.json
+++ b/versions/l-/libjpeg-turbo.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "7373af20f73237f0453fb02a1ba097e5f710d2f5",
+      "version": "2.1.3",
+      "port-version": 1
+    },
+    {
       "git-tree": "987bbe2533672eff1aaf160f0f36c818a1f986ee",
       "version": "2.1.3",
       "port-version": 0

